cpu/mips32r2_common/newlib_syscalls_mips_uhi/syscalls.c

```diff
@@ -46,6 +46,8 @@ extern char _sheap; /* start of the heap */
 extern char _eheap; /* end of the heap */
 char *heap_top = &_sheap + 4;
 
+/* MIPS newlib crt implements _init,_fini and _exit and manages the heap */
+#ifndef __mips__
```

Comment:
@cladmi do you mean that this should be forked off?

Reply:
If possible yes, also with the related changes in `Makefile.dep`.

I think it would would help comparing the size change between master and this one as it is my main way of debugging currently.